### PR TITLE
test(docs): testing doc snippets proof of concept

### PR DESCRIPTION
**Pull request by d0choa**
This PR goes against `il-howtos` and presents a proof of concept to test the python snippets in the documentation using pytest.

### CODE REVIEW

File: tests/docs/test_creating_spark_session.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+
+def test_default_session() -> None:
+    """Test default session."""
+    session = default_session()
+    assert isinstance(session, Session)
+
+
+def test_custom_session() -> None:
```
Comment (d0choa) [inline tests/docs/test_creating_spark_session.py:16]: I tried an assert to evaluate that the session is using 4G and it isn't. I think `session` contains the pytest session (probably through `getOrCreate`).

I'm not too concern about this, just saying is not an extremely canonical test, but it should be able to pick if something changes with the Session class.

Comment (DSuveges) [reply tests/docs/test_creating_spark_session.py:16]: So the code inside the how-tos are filled from an actual python scripts that are tested. Great.

**Review by ireneisdoomed — APPROVED** [review]
This approach entails a considerable effort (externalise the snippets to a separate module, think how to organise them to duplicate as little as possible, and finally test these functions in another module)... but the truth is that despite the simple documentation we had, I have already caught a couple of bugs.
I don't think the tutorials section is going to be very vast, so I think it's worth it and makes the documentation very solid.

**Comment by DSuveges** [comment]
> This approach entails a considerable effort (externalise the snippets to a separate module, think how to organise them to duplicate as little as possible, and finally test these functions in another module)

If we consider providing a examples and how-tos, we should really make sure these documenations won't get obsoleted. For this purpose the above solution works.

**Pull request merged**